bin/cron/applab_datasets/covid19

```diff
@@ -0,0 +1,128 @@
+
+  # Group records by date
+  daily_data = {}
+  lines.each do |line|
+    record = {}
+    parsed_date = Date.strptime(line[0], "%Y-%m-%d")
+    formatted_date = parsed_date.strftime('%a %-m/%-d') # ex Wed 1/7
+    record['Date'] = formatted_date
+    record['State'] = line[1]
+    # skip column 2 (FIPS code)
+    record['Confirmed New Cases'] = line[3].to_i
+    record['Deaths'] = line[4].to_i
+    if daily_data[formatted_date].nil?
+      daily_data[formatted_date] = []
+    end
+    daily_data[formatted_date].push record
+  end
+
+  # Truncate to MAX_DAYS most recent days
+  days = if daily_data.length > MAX_DAYS
+           daily_data.keys.slice(daily_data.length - MAX_DAYS, MAX_DAYS)
+         else
+           daily_data.keys
+         end
+
+  # Add id to each record
+  columns = ['id', 'Date', 'State', 'Confirmed New Cases', 'Deaths']
+  records = {}
+  id = 1
+  days.each do |day|
+    day_records = daily_data[day]
+    day_records.each do |record|
+      record['id'] = id
+      records[id] = record.to_json
+      id += 1
+    end
+  end
+
+  return records, columns
+end
+
+def get_world_covid19_data
+  package = DataPackage::Package.new('https://datahub.io/core/covid-19/datapackage.json')
+  resource = package.resources.find {|r| r['name'] == "countries-aggregated_csv"}
+  return unless resource
+  response = Net::HTTP.get_response(URI(resource['path']))
+  return unless response.code == '200'
+  csv = CSV.new(response.body)
+  lines = csv.read
+  lines.shift #first line is column names
+  weekly_data = {}
+
+  # Aggregate daily data by week
+  lines.each do |line|
+    parsed_date = Date.strptime(line[0], "%Y-%m-%d")
+    if weekly_data[parsed_date.cweek].nil?
+      weekly_data[parsed_date.cweek] = {}
+    end
+    if weekly_data[parsed_date.cweek][line[1]].nil?
+      weekly_data[parsed_date.cweek][line[1]] = {confirmed: 0, recovered: 0, deaths: 0}
+    end
+    country = weekly_data[parsed_date.cweek][line[1]]
+    country[:confirmed] += line[2].to_i
+    country[:recovered] += line[3].to_i
+    country[:deaths] += line[4].to_i
+  end
+
+  # Truncate to MAX_WEEKS most recent weeks
+  weeks = if weekly_data.length > MAX_WEEKS
+            weekly_data.keys.slice(weekly_data.length - MAX_WEEKS, MAX_WEEKS)
+          else
+            weekly_data.keys
+          end
+
+  columns = ['id', 'Date', 'Country', 'Confirmed New Cases', 'Recovered', 'Deaths']
+  records = {}
+  id = 1
+  weeks.each do |week_number|
+    date = Date.commercial(2020, week_number)
+    countries = weekly_data[week_number]
+    countries.each do |country, counts|
+      record = {}
+      record['id'] = id
+      record['Date'] = date.strftime('%a %-m/%-d') # ex Wed 1/7
+      record['Country'] = country
+      record['Confirmed New Cases'] = counts[:confirmed]
+      record['Recovered'] = counts[:recovered]
+      record['Deaths'] = counts[:deaths]
+
+      records[id] = record.to_json
+      id += 1
+    end
+  end
+  return records, columns
+end
```

Comment:
NIT: Some of this looks duplicated from states. Does it make sense to factor any of it out?

Reply:
I think it makes sense to keep them separate for clarity. The data are similar, but not the same, so I think it would end up being fairly confusing if we tried to re-use the code.